### PR TITLE
Regenerate Appeals rswag docs with OAuth details

### PR DESCRIPTION
**Pull request by caseywilliams**
## Description of change
This updates the rswag specs so that they correctly use an OAuth token when making requests to endpoints in the segmented appeals APIs (but not in decision reviews). This brings in many updates to example responses due to changes previously made in fixtures - these changes do not relate to OAuth, but have been pending for a while.

## Things to know about this PR

Sorry this diff is so huge. Here are some notes:

### OAuth-related changes

* The new `with_rswag_auth` helper used in each example is a thin wrapper around the existing `with_openid_auth` helper, but it's conditional based on which API docs are being built (no-op for decision reviews).
* Like other headers in rswag specs, we need to define the `Authorization` header using `let` instead of when we're making the request
* The configured `productionOauth`, `sandboxOauth`, and `bearer_token` security schemes were copied from the equivalent values in the claims API. For comparison, here's their [rswag config](https://github.com/department-of-veterans-affairs/vets-api/blob/475c47e1a865a484a9fc3015b87b362dd7736e7f/modules/claims_api/spec/support/rswag_config.rb#L34-L69), an example [rendered JSON file](https://github.com/department-of-veterans-affairs/vets-api/blob/475c47e1a865a484a9fc3015b87b362dd7736e7f/modules/claims_api/app/swagger/claims_api/v2/swagger.json#L64-L78), and [the web UI](https://developer.va.gov/explore/benefits/docs/claims) with OAuth.
* I think we'll need more examples for OAuth failures before we're done updating these docs, but I figured I'd add those in a different PR.

### Other changes in rendered JSON

* Non-veteran claimant and alternate signer headers have been introduced (from recent development work)
* `v2` has become `v0` in segmented APIs
* Shared schemas are introduced to segmented APIs
* `description`s have changed in a few places where `PUT` happens because the `put_description.md` file was "anonymized" a bit with a wildcard `*` in #10737 so that it could be reused
* A few example values have been removed - I think I remember that we removed some examples on purpose recently, but let me know if that's not right.
* Error codes, details, and sources change in several places - I am assuming these are due to updates in fixtures - this is the part I'm least sure about.

### CODE REVIEW

File: modules/appeals_api/app/swagger/higher_level_reviews/v0/swagger_dev.json [hunk 1]
```diff
@@ -34,7 +49,6 @@
             "name": "X-VA-SSN",
             "required": true,
             "description": "Veteran's SSN",
-            "example": "706547821",
```
Comment (cilestin) [inline modules/appeals_api/app/swagger/higher_level_reviews/v0/swagger_dev.json:37]: I think we want to keep these examples, as they were recently updated in @michelpmcdonald's ticket to update on LOAST findings.

File: modules/appeals_api/app/swagger/notice_of_disagreements/v0/swagger_dev.json [hunk 2]
```diff
@@ -699,48 +739,6 @@
                                     ]
                                   }
                                 },
-                                "claimant": {
```
Comment (cilestin) [inline modules/appeals_api/app/swagger/notice_of_disagreements/v0/swagger_dev.json:702]: This field shouldn't have been removed - it's present in all our v2 10182 files, so I'm not sure why it disappeared here...

File: modules/appeals_api/app/swagger/notice_of_disagreements/v0/swagger_dev.json [hunk 3]
```diff
@@ -2598,6 +2635,39 @@
         "type": "apiKey",
```
Comment (cilestin) [inline modules/appeals_api/app/swagger/notice_of_disagreements/v0/swagger_dev.json:2635]: Doesn't this need to change to oauth too?

File: modules/appeals_api/spec/support/doc_helpers.rb [hunk 4]
```diff
@@ -24,14 +24,34 @@ def normalize_evidence_submission_response(response)
...
+    return config if DocHelpers.decision_reviews?
+
+    config + [{ productionOauth: oauth_scopes }, { sandboxOauth: oauth_scopes }, { bearer_token: [] }]
+  end
+
   # @param [Hash] opts
   # @option opts [String] :desc The description of the test. Required.
   # @option opts [Symbol] :response_wrapper Method name to wrap the response, to modify the output of the example
   # @option opts [Boolean] :extract_desc Whether to use the example name
   # @option opts [Boolean] :skip_match Whether to skip the match metadata assertion
```
Comment (cilestin) [inline modules/appeals_api/spec/support/doc_helpers.rb:49]: Be sure to add `scopes` in the doc block :)

**Review by cilestin — CHANGES_REQUESTED** [review]
Lookin' really good - just some small things to fix up before merging.

File: modules/appeals_api/app/swagger/higher_level_reviews/v0/swagger_dev.json (hunk 1 above)
Comment (caseywilliams) [reply modules/appeals_api/app/swagger/higher_level_reviews/v0/swagger_dev.json]: It looks like the missing example values were previously just added to the JSON files (as opposed to the rswag specs themselves), so they all got overwritten once the JSON was regenerated - I think I've found a solution but I left a few comments with notes.

File: modules/appeals_api/app/swagger/appeals_api/v2/swagger.json [hunk 5]
```diff
@@ -316,6 +322,7 @@
             "name": "X-VA-SSN",
             "required": true,
             "description": "Veteran's SSN",
+            "example": "706547821",
```
[the remark this answers is not in the document]
Comment (cilestin) [reply modules/appeals_api/app/swagger/appeals_api/v2/swagger.json:325]: Yup I'm fine with it, too.

File: modules/appeals_api/spec/support/swagger_shared_components.rb [hunk 6]
```diff
@@ -34,14 +34,16 @@ def self.header_params
         type: :string,
         name: 'X-VA-SSN',
         required: true,
-        description: 'Veteran\'s SSN'
+        description: 'Veteran\'s SSN',
+        example: '706547821'
```
[the remark this answers is not in the document]
Comment (cilestin) [reply modules/appeals_api/spec/support/swagger_shared_components.rb:38]: I think keeping one example user global is fine. We may show off alternatives in examples at some point, but I don't think that'll change this code in any appreciable way regardless.

**Review by cilestin — APPROVED** [review]

**Pull request merged**